extensions/amp-video/1.0/storybook/Basic.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../../src/preact';
+import {VideoWrapper} from '../video-wrapper';
+import {boolean, object, text, withKnobs} from '@storybook/addon-knobs';
+import {withA11y} from '@storybook/addon-a11y';
+
+const {forwardRef, useRef, useImperativeHandle} = Preact;
+
+export default {
+  title: 'Video Wrapper',
+  component: VideoWrapper,
+  decorators: [withA11y, withKnobs],
+};
+
+const VideoTagPlayer = forwardRef((props, ref) => {
+  const {
+    children,
+
+    src,
+    muted,
+    loop,
+    controls,
+    poster,
+
+    'aria-label': ariaLabel,
+    album,
+    artist,
+    artwork,
+    title,
+
+    onLoad,
+    onLoadedMetadata,
+    onPause,
+    onPlaying,
+  } = props;
+
+  const videoNodeRef = useRef();
+
+  useImperativeHandle(ref, () => ({
+    play: () => videoNodeRef.current.play(),
+    pause: () => videoNodeRef.current.pause(),
+    getMetadata: () => ({
+      title: title || '',
+      artist: artist || '',
+      album: album || '',
+      artwork: [{src: artwork || poster || ''}],
+    }),
+  }));
+
+  return (
+    <video
+      ref={videoNodeRef}
+      aria-label={ariaLabel}
+      src={src}
+      muted={muted}
+      loop={loop}
+      controls={controls}
+      onPlaying={onPlaying}
+      onPause={onPause}
+      poster={poster}
+      style={{position: 'relative', width: '100%', height: '100%'}}
+      onLoadedMetadata={() => {
+        onLoad();
```

Comment:
Trigger `load` event on `loadedmetadata` like the current `loadPromise` implementation:

https://github.com/ampproject/amphtml/blob/b6313e372fdd1298928e2417dcc616b03288e051/src/event-helper.js#L183-L194

@gmajoulet might have more context on why we use that event.

Reply:
Not much context unfortunately. `loadedmetadata` is the fastest you can resolve/reject the `layoutCallback()` reliably (`loadstart` doesn't have a way to synchronously check its status).
If you want to revisit which event to use: the only thing that's waiting for `layoutCallback` to resolve is the poster that gets removed.

That gives two possible experiences:

1. We want the poster until the first frame is available: should use `loadeddata`
2. We want the poster until the video can play: should use `canplay`

Reply:
Ah, good to know. In that case then we only care about the `AmpVideo` replacement implementation, not this placeholder for the sample.